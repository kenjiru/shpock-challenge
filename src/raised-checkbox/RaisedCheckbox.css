.raised-checkbox {
    height: 30px;
}

.raised-checkbox input[type=checkbox] + div {
    align-items: center;
}