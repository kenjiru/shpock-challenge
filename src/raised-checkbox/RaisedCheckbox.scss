.raised-checkbox {
    height: 30px;

    input[type=checkbox] + div {
        align-items: center;
    }
}
