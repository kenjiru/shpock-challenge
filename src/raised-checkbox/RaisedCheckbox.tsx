import * as classNames from "classnames";
import * as React from "react";
import { Component, CSSProperties, ReactElement } from "react";
import { Checkbox, CheckboxProps } from "material-ui";
import { grey700 } from "material-ui/styles/colors";

import RaisedContainer from "../raised-container/RaisedContainer";

import "./RaisedCheckbox.scss";

class RaisedCheckbox extends Component<CheckboxProps> {
    public static CLASS_NAME: string = "raised-checkbox";
    public static LINE_HEIGHT: string = "18px";
    public static CHECKED_COLOR: string = "white";
    public static NORMAL_COLOR: string = grey700;
    public static SIZE: number = 20;
    public static FONT_SIZE: number = 13;

    public render(): ReactElement<HTMLElement> {
        return (
            <RaisedContainer
                className={this.getClassName()}
                checked={this.props.checked}
            >
                <Checkbox
                    {...this.props}
                    iconStyle={this.getIconStyle()}
                    labelStyle={this.getLabelStyle()}
                />
            </RaisedContainer>
        );
    }

    private getIconStyle(): CSSProperties {
        const {checked} = this.props;

        return {
            fill: checked ? RaisedCheckbox.CHECKED_COLOR : RaisedCheckbox.NORMAL_COLOR,
            width: RaisedCheckbox.SIZE,
            height: RaisedCheckbox.SIZE
        };
    }

    private getLabelStyle(): CSSProperties {
        const {checked} = this.props;

        return {
            color: checked ? RaisedCheckbox.CHECKED_COLOR : RaisedCheckbox.NORMAL_COLOR,
            fontSize: RaisedCheckbox.FONT_SIZE,
            lineHeight: RaisedCheckbox.LINE_HEIGHT
        };
    }

    private getClassName(): string {
        return classNames(RaisedCheckbox.CLASS_NAME, this.props.className);
    }
}

export default RaisedCheckbox;
