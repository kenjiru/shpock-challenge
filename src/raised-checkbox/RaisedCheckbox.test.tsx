import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import shallowToJson from "enzyme-to-json";
import { Checkbox, CheckboxProps } from "material-ui";

import RaisedCheckbox from "./RaisedCheckbox";

describe("RaisedCheckbox", () => {
    it("should render correctly", () => {
        const output: ShallowWrapper = shallow(
            <RaisedCheckbox/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should pass the correct styles to the underlying Checkbox", () => {
        const output: ShallowWrapper = shallow(
            <RaisedCheckbox/>
        );

        let props: CheckboxProps = output.find(Checkbox).props();

        expect(props.iconStyle.width).toBe(RaisedCheckbox.SIZE);
        expect(props.iconStyle.height).toBe(RaisedCheckbox.SIZE);
        expect(props.labelStyle.lineHeight).toBe(RaisedCheckbox.LINE_HEIGHT);
    });

    it("should have the correct color when it's checked", () => {
        const output: ShallowWrapper = shallow(
            <RaisedCheckbox checked={true}/>
        );

        let props: CheckboxProps = output.find(Checkbox).props();

        expect(props.iconStyle.fill).toBe(RaisedCheckbox.CHECKED_COLOR);
        expect(props.labelStyle.color).toBe(RaisedCheckbox.CHECKED_COLOR);
    });

    it("should have the correct color when it's NOT checked", () => {
        const output: ShallowWrapper = shallow(
            <RaisedCheckbox checked={false}/>
        );

        let props: CheckboxProps = output.find(Checkbox).props();

        expect(props.iconStyle.fill).toBe(RaisedCheckbox.NORMAL_COLOR);
        expect(props.labelStyle.color).toBe(RaisedCheckbox.NORMAL_COLOR);
    });
});
