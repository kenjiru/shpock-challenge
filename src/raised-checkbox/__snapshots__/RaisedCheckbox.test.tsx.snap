// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaisedCheckbox should render correctly 1`] = `
<MuiComponent
  className="raised-checkbox"
>
  <Checkbox
    disabled={false}
    iconStyle={
      Object {
        "fill": "#616161",
        "height": 20,
        "width": 20,
      }
    }
    labelPosition="right"
    labelStyle={
      Object {
        "color": "#616161",
        "fontSize": 13,
        "lineHeight": "18px",
      }
    }
  />
</MuiComponent>
`;
